src/setup-beam.js

```diff
@@ -55,38 +49,44 @@ async function main() {
   await maybeInstallRebar3(rebar3Spec)
 }
 
-async function installOTP(otpSpec, osVersion, hexMirrors) {
-  const otpVersion = await getOTPVersion(otpSpec, osVersion, hexMirrors)
+async function installOTP(otpSpec, osVersion) {
+  const otpVersion = await getOTPVersion(otpSpec, osVersion)
   core.startGroup(`Installing Erlang/OTP ${otpVersion} - built on ${osVersion}`)
-  await installer.installOTP(osVersion, otpVersion, hexMirrors)
+  await doWithMirrors({
+    hexMirrors: hexMirrorsInput(),
+    actionTitle: `install Erlang/OTP ${otpVersion}`,
+    action: async (hexMirror) => {
+      await install('otp', {
+        osVersion,
+        toolVersion: otpVersion,
+        hexMirror,
+      })
+    },
+  })
```

Comment:
This is what we did to replace input `hexMirrors` with internal function iteration: we declare a function that implements the iteration internally (which hopefully eases maintenance and makes for more consistent behaviour, when required).